### ISSUE TITLE
sdk/state: change horizon ingestion to only ingest txs for the escrow accounts

### ISSUE THREAD

**Issue by leighmcculloch** [stellar/experimental-payment-channels#315]
When I implemented Horizon ingestion I initially made it stream transactions that affected the escrow accounts, but this resulted in out-of-order and duplicate transactions which was a problem with the state packages ingestion code at the time, so I changed the Horizon ingestion to simply stream all transactions on the whole network which ensured they would all be streamed in order.

With the changes @acharb is making for #267 (#312) and that we plan to make for #314, it will be possible to ingest transactions out of order and to ingest duplicates. Once those issues are closed we can change Horizon ingestion back to stream transactions from only the escrow accounts.

Dependent on #267 #314

**Comment by leighmcculloch** [comment]
Duplicate of #320.
